### ISSUE TITLE
gjtech.net is still linked to in the metadata, which it probably shouldn't be

### ISSUE THREAD

**Issue by DandelionSprout** [AdAway/adaway.github.io#24]
The lines:

```
# 2013-03-29 Integrated some hosts from
#            http://adblock.gjtech.net/?format=hostfile
```

...are still present in the new AdAway list, despite how `gjtech.net` has for several years now been a parked domain with no actual lists and a not insignificant chance (Maybe as much as 20%) of being redirected to malware, and is probably *not* a link that laymen end-users should want to casually check out. I therefore hereby request that the text be replaced with `# 2013-03-29 Integrated some hosts from GJtech`.

The rest of the metadata in the list's top could also need a bit of updating to whatever you (jawz101) and the AdAway programmer team would want to have. In particular I personally presume that it could be worth it to encourage end-users to report list-only issues to https://github.com/AdAway/adaway.github.io/issues.

I would also like to thank the AdAway team in general for **finally** removing the `gjtech.net` link from https://github.com/AdAway/AdAway/wiki/HostsSources, even if it was done perhaps several years late.

**Comment by PerfectSlayer** [comment]
Hello @DandelionSprout 

Thanks for your report. I agree the hosts file metadata need to be updated. 

First, contribution links should be better if linked to this repo (and not the app repo).
Then, changelog could be removed from the hosts file.
It is part of the git log and some information are wrong. I mean, I removed clould flare month ago for example.

If a dedicated human friendly changelog is required, I would prefer it to be stored in a `changelog.md` file rather than in the hosts file itself.
About metadata, the minimum needed for me is:
* Name, description, original URL,
* License information,
* Contributors
* How to contribute / GitHub link

@jawz101 What do you think about it? Would like to have a look at it some time?

**Comment by jawz101** [comment]
How about this.  I looked at the Creative Commons license in the link and it sounds like you have to list contributors and changes.  I'm all for taking stuff out and not maintaining a handtyped list of changes or contributors.

```
# AdAway default blocklist
# Blocking mobile ad providers and some analytics providers
#
# Project home page:
# https://github.com/AdAway/adaway.github.io/
#
# Fetch the latest version of this file:
# https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt
#
# License:
# CC Attribution 3.0 (http://creativecommons.org/licenses/by/3.0/)
#
# Contributions by:
# Kicelo, Dominik Schuermann.  Further changes and contributors maintained in the commit history at https://github.com/AdAway/adaway.github.io/commits/master
#
# Contribute:
# Create an issue at https://github.com/AdAway/adaway.github.io/issues
#

127.0.0.1 localhost
127.0.0.1 localhost.localdomain
127.0.0.1 local
255.255.255.255 broadcasthost
::1 localhost
::1 ip6-localhost
::1 ip6-loopback
fe80::1%lo0 localhost
ff00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts
```

**Comment by jawz101** [comment]
What do you all think of that header?

**Comment by spirillen** [comment]
That's a nice header :smiley: But I would suggest a little change.

```
# Contributions by:
# Kicelo, Dominik Schuermann.
# Further changes and contributors maintained.....
```

Add the 3 - 5 most active contributers _OR_ Those who have a decision right (merge). I this way everybody would know who to rely on and contact on the issue board in case of 'what ever'

**Comment by jawz101** [comment]
Well, they are directed here to report issues.  I just kept the people on the list who originally documented themselves and the rest I just wanted to keep static with a pointer to the repo & changelog.  Personally, I don't like getting my identity out there much.

As for Kicelo, I don't know if that is a github user but I cannot see a github user with that name and I don't have interest in tracking down their real identity.

**Comment by spirillen** [comment]
Good point :+1:

**Comment by PerfectSlayer** [comment]
@jawz101 The header is good. Thanks for the update.

And yes, don't loose too much time duplicating content (I mean adding stuff to the header that is already easily available on the GitHub).

And to limit history to the hosts file only, you can link to: https://github.com/AdAway/adaway.github.io/commits/master/hosts.txt (with hosts.txt at the end of URL) 😉

**Comment by spirillen** [comment]
There is a license conflict

https://github.com/AdAway/adaway.github.io/blob/8e7d98dfaaa88abbe8fd4778b22ea7c9c9e87525/hosts.txt#L11

Says `# CC Attribution 3.0 (http://creativecommons.org/licenses/by/3.0/)`

But the `https://github.com/AdAway/adaway.github.io/blob/master/LICENSE.md` is `GNU General Public License v3.0`

Which is the correct one? Alternatively it should be stated that file x is license x while file y is license y

**Comment by DandelionSprout** [comment]
I presume this would've worked better as a separate issue report, but I can just as well go along with it for now.

In the event that the AdAway team opts for CC Attribution 3.0 or for dual-licencing, I presume the in-list link should be upgraded to HTTPS, i.e. https://creativecommons.org/licenses/by/3.0/.

**Comment by jawz101** [comment]
I just left the license on the host file the same as it was before I did the change.  I don't know if it's up to me, @dschuermann, @PerfectSlayer, or anyone to make that determination.  Looking into the Creative Commons license it might be the most appropriate since it reads like it is intended for something like a creative work rather than a piece of software.  I think of Adaway _the app_ as a piece of software (more GPL in nature) and a list of domains to block as a work of creativity so I could see why it might carry a different license just for the blocklist file.

**Comment by DandelionSprout** [comment]
I remember creating a licence *specifically* designed for adblock list makers (although with only a so-so weight from a presumed lawyer perspective) called the [Dandelicence](https://raw.githubusercontent.com/DandelionSprout/Dandelicence/master/DandelicenceV1.2.md). Although designed so that compilation list makers would be far less likely to receive flatout DMCA takedowns from GPLv3 projects (by virtue of encapsulating GPLv3 content so that it remains under GPLv3 conditions), it also allows specifying individual licences for each file (which I think is what you'd want to do here), a provision that I'm pretty doesn't exist in vanilla GPLv3.

**Comment by jawz101** [comment]
my MobileAdTrackers license was the MIT license with a change that essentially said if you make millions from it I would like some free stuff.

**Comment by PerfectSlayer** [comment]
@jawz101 It's a touchy question here. Licensing is not an easy topic.
@dschuermann is the one who add the hosts file to this repository. I guess it is also the one who creates it. Before GitHub, it previously was hosted on adaway.sufficientlysecure.org which is his website.

@DandelionSprout Who do you think will issue DMCA takedowns? Listmakers to listmakers or company to listmakers?

**Comment by DandelionSprout** [comment]
I don't think the DMCA question will actually be relevant to you guys' project, but in my case it was inspired chiefly by some listmaker-to-listmaker takedown instances, as I know that some subcategories of tech enthusiasts can be pretty dedicated to their GPLv3 licence worship.

Of particular note in the past 3 years, there's been 2 big cases where
* Energized Protection came within arms' reach of being taken down by Steven Black before the former changed its licence (The GitHub thread about it seems to me to have been deleted in connection to a repo URL change).
* `CHEF-KOCH` (who should **not** be mentioned outside of code quotes for several reasons) got a creditation request by the maker of WindowsSpyBlocker; but instead of performing the friendly favour, chose to double down on saying he made the list on his own, leading to WindowsSpyBlocker's maker issuing [a DMCA that is generally considered to be rightful](https://github.com/github/dmca/blob/master/2019/03/2019-03-05-WindowsSpyBlocker.md).

This led me to create a licence which can be described as *"It's okay to make mistakes, as long as we're all friendly to each other!"*

**Comment by spirillen** [comment]
> I presume this would've worked better as a separate issue report, but I can just as well go along with it for now.
> 


Hey @DandelionSprout th reason I posted the very question here, is that the header implanted into the hosts file came from this post. No more, no less, as I considered it as a mistake from @jawz101 :smile:

**Comment by jawz101** [comment]
@spirillen before I touched it, this was what it looked like:

```
# AdAway default blocklist
# Blocking mobile ad providers and some analytics providers
# 
# Contribute:
# Create an issue at https://github.com/AdAway/AdAway/issues
#
# Changelog:
# 2016-07-18 Now hosted on GitHub + Cloudflare
# 2014-05-18 Now with a valid SSL certificate available at https://adaway.org/hosts.txt
# 2013-03-29 Integrated some hosts from
#            http://adblock.gjtech.net/?format=hostfile
# 2013-03-14 Back from the dead
# 
# License:
# CC Attribution 3.0 (http://creativecommons.org/licenses/by/3.0/)
#
# Contributions by:
# Kicelo, Dominik Schuermann
#
```

All I did was take the suggestions from this thread and just reword it to remove the changelog portion and just have it point back to the repo and leave all of the original contributors.  I didn't change what the license was because that's what it was before I touched it.

```
# AdAway default blocklist
# Blocking mobile ad providers and some analytics providers
#
# Project home page:
# https://github.com/AdAway/adaway.github.io/
#
# Fetch the latest version of this file:
# https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt
#
# License:
# CC Attribution 3.0 (http://creativecommons.org/licenses/by/3.0/)
#
# Contributions by:
# Kicelo, Dominik Schuermann.
# Further changes and contributors maintained in the commit history at
# https://github.com/AdAway/adaway.github.io/commits/master
#
# Contribute:
# Create an issue at https://github.com/AdAway/adaway.github.io/issues
#
```

**Comment by spirillen** [comment]
@jawz101 I still toke it as a `copy/paste` at the time I posted the license mismatch, haven't done the big investigation :smile:
